st3/lsp_utils/generic_client_handler.py

```diff
@@ -40,9 +41,14 @@ def setup(cls) -> None:
 
     @classmethod
     def cleanup(cls) -> None:
+
+        def run_async() -> None:
+            if os.path.isdir(cls.package_storage()):
+                shutil.rmtree(cls.package_storage())
+
+        if events.remove(cls.package_name):
+            sublime.set_timeout_async(run_async, 1000)
```

Comment:
Thanks for that!

I don't trust arbitrary timeouts though. Would probably try to add some API to LSP to notify when the session has ended.

Reply:
The session is not necessarily started at this point though...

Reply:
Can we use it as a temp solution？

Reply:
I will look into it more soon, when I have free time. Might go with your solution if I can't find anything better?

What are your exact testing steps on Windows?

Reply:
So after checking stuff, I'm even less happy about that arbitrary timeout because there is no guarantee that the session will shutdown in 1s. It could easily take longer.

I guess this has a chance of improving situation a bit so I'm not against it but I wonder if we should investigate a more proper solution for this. Especially considering that @rwols has some plans for it for the 3.8 runtime already (which IMO is still quite far away).

Reply:
So this is the case when package is uninstalled.

When package gets updated, we will no longer delete the files on disabling the package with the idea that the server will get updated on starting the session after updating. But in that case it could also happen that the old session hasn't shut down yet when the new session is starting. This is even less likely, I guess, since updates typically take more time and so the server has more time to shut down. So it's probably OK to just merge it for now.